### PR TITLE
Fix Claude detection reliability with child process discovery

### LINKED ISSUE #17: Claude detection fails to find running instances in tmux sessions
## Problem
The current Claude detection logic fails to detect running Claude instances in tmux sessions. Users report that Claude is clearly running and visible in their tmux panes, but the TUI shows no detected instances.

## Root Cause Investigation
Through spike investigation, discovered that:

1. **Process hierarchy issue**: tmux reports the shell process (typically `node`) as the pane's current command
2. **Child process reality**: Claude actually runs as a child process under that shell (e.g., `node:5486` → `claude:6117`)
3. **Detection gap**: Current logic only checks the reported shell process, missing the actual Claude child process

## Technical Details
- Claude instances commonly run as `node` processes with `claude` as child
- tmux `pane_current_command` shows parent shell, not the actual running application
- Need to traverse process tree to find child processes

## Solution Approach
- Add child process discovery using `ps` command
- Enhance detection logic to check both parent and child processes
- Improve content-based fallback detection
- Add comprehensive E2E testing with real tmux sessions

## Impact
- Users unable to use auto-yes functionality for running Claude instances
- False negatives leading to missed automation opportunities
- Poor user experience when detection fails

## Acceptance Criteria
- [ ] Detection finds Claude instances running as child processes
- [ ] E2E tests validate detection with real tmux sessions
- [ ] No false positives from reports or logs mentioning Claude
- [ ] Maintains backward compatibility with existing detection patterns

### PR DESCRIPTION
**Pull request by safurrier**
## Summary
Fixed Claude detection failing to find running instances in tmux sessions by implementing child process discovery.

## Problem
Detection was missing Claude instances because it only checked the shell process (node) reported by tmux, but Claude actually runs as a child process under that shell.

## Changes

**Child Process Discovery**
- Added `find_child_processes()` method to discover child processes using `ps` command
- Enhanced `is_claude_process()` to check both parent and child processes for Claude instances
- Handles common case where tmux reports shell but Claude runs as child

**Content Detection Improvements**  
- Made content detection more specific to avoid false positives
- Added detection for "Welcome to Claude Code" screen
- Requires additional Claude-specific patterns (cwd:, /help, Tip:) to avoid detecting reports/logs

**Comprehensive Testing**
- Created real tmux E2E tests with context managers for proper cleanup
- Added integration tests for child process discovery with mocked process trees
- Unit tests for enhanced detection methods and false positive prevention

## Technical Details

**Process Hierarchy Discovery:**
```python
def find_child_processes(self, parent_pid: str) -> list[dict[str, str]]:
    # Uses ps -eo pid,ppid,command to find children
    # Returns list of child processes with command info
```

**Enhanced Detection Logic:**
- First checks direct process command (existing behavior)
- Then discovers and checks child processes for Claude instances
- Maintains backward compatibility with current detection patterns

## Testing
**Manual testing:**
- Created real tmux sessions with Claude instances
- Verified detection works for both direct and child process scenarios
- Tested false positive prevention with Claude reports/logs

**Automated testing:**
- E2E tests with real tmux session management
- Integration tests with mocked process trees
- Unit tests covering new detection methods

**Test scenarios validated:**
- Claude running as direct command in tmux pane ✓
- Claude running as child process under node shell ✓  
- Content-based detection for welcome screens ✓
- False positive prevention for reports mentioning Claude ✓

Closes #17

### CODE REVIEW

File: claude_code_autoyes/core/detector.py [hunk 1]
```diff
@@ -114,6 +178,19 @@ def is_claude_pane(self, content: str) -> bool:
...
+            return True
+
+        # Enhanced: Claude interface with box patterns (even without cursor)
+        # Be more specific to avoid false positives from reports/logs about Claude
+        if (
+            "Claude Code" in content
+            and "╰─" in content
+            and ("cwd:" in content or "/help for help" in content or "Tip:" in content)
+        ):
+            return True
```
Comment (safurrier) [inline claude_code_autoyes/core/detector.py]: This will basically identify the startup screen, but potentially not other non start up screens. 

It's likely hard to capture possible screen just by the text on screen.

But I think we should refactor this function to be more modular for matching the various patterns b/c right now it's pretty long and it's a bit of a code smell.

File: tests/integration/test_child_process_discovery.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
+"""Integration tests for child process discovery functionality."""
```
Comment (safurrier) [inline tests/integration/test_child_process_discovery.py]: Are these tests actually useful given all the mocking? Seems like it might be b/c of the logic getting tested but I want to confirm. Explain to me how it's useful or not.

File: tests/unit/test_enhanced_detector_methods.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
+"""Unit tests for enhanced ClaudeDetector methods."""
```
Comment (safurrier) [inline tests/unit/test_enhanced_detector_methods.py]: Same with this (if it's valuable) seems like it just for debugging purposes over like true tests? But it could scale if we run into more edge cases.

File: conventions_research_report.md
Comment (safurrier) [inline conventions_research_report.md:1]: Consolidate this with the file PROJECT_CONVENTIONS.md

If that file needs to be updated with anything from this, then delete this file.
